crates/adapters/src/server/mod.rs

```diff
@@ -388,59 +502,96 @@ pub fn run_server(
         }
     }
 
+    #[cfg(not(feature = "feldera-enterprise"))]
+    if config.global.fault_tolerance.is_enabled() {
+        return Err(ControllerError::EnterpriseFeature("fault tolerance"));
+    }
+
+    // Initiate creating the controller so that we can get access to storage,
+    // which is needed to determine the initial state.
+    let builder = ControllerBuilder::new(&config)?;
+    let initial_status = builder
+        .storage()
+        .and_then(|storage| StoredStatus::read(&*storage))
+        .map_or(args.initial, |s| s.desired_status);
```

Comment:
Shouldn't this be conditional on the deployment ID being different?
If it is different, it should use the one from initial arg.
If it is not different, it should use the one from storage.